tools/mcstas_to_nexus.ipynb

```diff
@@ -0,0 +1,219 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "0",
+   "metadata": {},
+   "source": [
+    "# McStas to NeXus for ODIN\n",
+    "\n",
+    "This notebook converts data from a McStas simulation output (`.h5` filetype) to a NeXus file that uses a file for the Odin instrument (written by CODA) as a template for the geometry information.\n",
+    "\n",
+    "It adds events to the `timepix3` detector and the `beam_monitor_3` monitor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import scipp as sc\n",
+    "import scippnexus as sx\n",
+    "import h5py as h5\n",
+    "import shutil"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_mcstas_simulation_data(\n",
+    "    file_path,\n",
+    "    data_path=\"entry1/data/transmission_event_signal_dat_list_p_t_x_y_z_vx_vy_vz/events\",\n",
+    "    probability_scale_factor=1000.0,\n",
+    "):\n",
+    "    with sx.File(file_path, \"r\") as f:\n",
+    "        # The name p_t_x_y_z_vx_vy_vz represents\n",
+    "        # probability, time of arrival, position(x, y, z) and velocity(vx, vy, vz).\n",
+    "        # The name also represents the order of each field in the table.\n",
+    "        # For example, probability is the first field, so data['dim_1', 0] is the probability.\n",
+    "        data = f[data_path][()].rename_dims({\"dim_0\": \"event\"})\n",
+    "        probabilities = data[\"dim_1\", 0].copy()\n",
+    "        probabilities.unit = \"dimensionless\"\n",
+    "        time_of_arrival = data[\"dim_1\", 1].copy()\n",
+    "        time_of_arrival.unit = \"s\"  # Hardcoded unit from the data.\n",
+    "        positions = data[\"dim_1\", 2:5]\n",
+    "        counts = (probabilities / probabilities.max()) * probability_scale_factor\n",
```

Comment:
hmmm I think in what we did below we are not taking into account the weights/probabilities of the mcstas events.....
We are giving a weight of 1 to everything?

Reply:
Now using `np.repeat` to duplicate the events to reproduce the mcstas event statistics.